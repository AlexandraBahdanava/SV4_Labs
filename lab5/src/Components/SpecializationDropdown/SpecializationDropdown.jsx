import React, { useState } from 'react';
import { FormControl, InputLabel, Select, MenuItem, Button } from '@mui/material';

function SpecializationDropdown({ selectedSpecialization, onSpecializationChange, onApplyClick }) {
    const [buttonClicked, setButtonClicked] = useState(false);

    const handleApplyClick = () => {
        setButtonClicked(true);
        onApplyClick();
    };

    const handleSpecializationChange = (value) => {
        setButtonClicked(false);
        onSpecializationChange(value);
    };

    const specializationOptions = [
        'Классический маникюр и педикюр',
        'Спа-маникюр и педикюр',
        'Гель-лак и шеллак маникюр',
    ];

    return (
        <div>
            <FormControl variant="outlined" style={{ minWidth: 200, marginRight: 10 }}>
                <InputLabel id="specialization-label">Специализация</InputLabel>
                <Select
                    labelId="specialization-label"
                    id="specialization-select"
                    value={selectedSpecialization}
                    onChange={(e) => handleSpecializationChange(e.target.value)}
                    label="Специализация"
                >
                    <MenuItem value="">Все</MenuItem>
                    {specializationOptions.map((option, index) => (
                        <MenuItem key={index} value={option}>
                            {option}
                        </MenuItem>
                    ))}
                </Select>
            </FormControl>
        </div>
    );
}

export default SpecializationDropdown;
