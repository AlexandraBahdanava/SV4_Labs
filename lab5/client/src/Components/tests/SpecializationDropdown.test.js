import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SpecializationDropdown from '../SpecializationDropdown';

describe('SpecializationDropdown', () => {
    it('renders correctly', () => {
        const { getByLabelText } = render(
          <SpecializationDropdown selectedSpecialization="" onSpecializationChange={() => {}} onApplyClick={() => {}} />
        );
      
        expect(getByLabelText('Специализация')).toBeTruthy();
      });


  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <SpecializationDropdown selectedSpecialization="" onSpecializationChange={() => {}} onApplyClick={() => {}} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles specialization change correctly', () => {
    const mockSpecializationChange = jest.fn();
    const { getByLabelText, getByText } = render(
      <SpecializationDropdown selectedSpecialization="" onSpecializationChange={mockSpecializationChange} onApplyClick={() => {}} />
    );
  
    // Открываем выпадающий список
    fireEvent.mouseDown(getByLabelText('Специализация'));
  
    // Выбираем значение в выпадающем списке
    fireEvent.click(getByText('Классический маникюр и педикюр'));
  
    expect(mockSpecializationChange).toHaveBeenCalledWith('Классический маникюр и педикюр');
  });
  
});
