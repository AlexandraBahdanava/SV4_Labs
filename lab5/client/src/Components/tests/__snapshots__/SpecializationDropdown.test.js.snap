// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpecializationDropdown matches the snapshot 1`] = `
<div>
  <div
    className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    style={
      Object {
        "marginRight": 10,
        "minWidth": 200,
      }
    }
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      id="specialization-label"
    >
      Специализация
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      onClick={[Function]}
    >
      <div
        aria-controls=":r0:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="specialization-label specialization-select"
        className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="specialization-select"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden={true}
        aria-invalid={false}
        className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        disabled={false}
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value=""
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-yjsfm1"
        >
          <span>
            Специализация
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
